Sources/Workspace/CheckoutState.swift

```diff
@@ -29,8 +29,8 @@ extension CheckoutState: CustomStringConvertible {
             return revision.identifier
         case .version(let version, _):
             return version.description
-        case .branch(let branch, _):
-            return branch
+        case .branch(let branch, let revision):
+            return "\(branch)#\(revision.identifier)"
```

Comment:
Is `#` a good separator here? I'm sure I have ever seen that use to demarcate a revision.

Reply:
Would you prefer `@` or something more explicit like `"\(branch) at \(revision)"`?

Reply:
I think I'd prefer `"\(branch) at \(revision)"` unless there's a common notation for this.

Reply:
Using parens for the revision and its first 7 symbols for `description` and full revision string in `debugDescription` per our previous discussion.